File: baseLib/src/main/java/me/ycdev/android/lib/common/utils/Preconditions.java
package me.ycdev.android.lib.common.utils;

@SuppressWarnings({"unused", "WeakerAccess"})
public class Preconditions {
    public static void checkMainThread() {
        if (!ThreadUtils.isMainThread()) {
            throw new RuntimeException("Not in main thread");
        }
    }

    public static void checkNonMainThread() {
        if (ThreadUtils.isMainThread()) {
            throw new RuntimeException("In main thread");
        }
    }

    public static void checkArgument(boolean expression) {
        if (!expression) {
            throw new IllegalArgumentException();
        }
    }

    public static <T> T checkNotNull(T object) {
        if (object == null) {
            throw new NullPointerException();
        }
        return object;
    }
}

File: baseLib/src/main/java/me/ycdev/android/lib/common/utils/Preconditions.kt
package me.ycdev.android.lib.common.utils

object Preconditions {
    fun checkMainThread() {
        if (!ThreadUtils.isMainThread) {
            throw RuntimeException("Not in main thread")
        }
    }

    fun checkNonMainThread() {
        if (ThreadUtils.isMainThread) {
            throw RuntimeException("In main thread")
        }
    }

    fun checkArgument(expression: Boolean) {
        if (!expression) {
            throw IllegalArgumentException()
        }
    }

    fun <T> checkNotNull(obj: T?): T {
        if (obj == null) {
            throw NullPointerException()
        }
        return obj
    }
}
